apps/gp2-server/test/data-providers/contentful/calendar.data-provider.test.ts

```diff
@@ -0,0 +1,316 @@
+    Promise.resolve(environmentMock);
+
+  const calendarDataProvider = new CalendarContentfulDataProvider(
+    contentfulGraphqlClientMock,
+    contentfulRestClientMock,
+  );
+
+  const contentfulGraphqlClientMockServer =
+    getContentfulGraphqlClientMockServer({
+      Calendars: () => getContentfulGraphqlCalendar(),
+    });
+
+  const calendarDataProviderMock = new CalendarContentfulDataProvider(
+    contentfulGraphqlClientMockServer,
+    contentfulRestClientMock,
+  );
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  const getContentfulCalendarDataObject = () => ({
+    ...getCalendarDataObject(),
+    ...calendarUnreadyResponse,
+  });
+
+  describe('Fetch method', () => {
+    test('Should fetch the list of calendars from Contentful GraphQl', async () => {
+      const result = await calendarDataProviderMock.fetch({});
+
+      expect(result.items[0]).toMatchObject(getContentfulCalendarDataObject());
+    });
+
+    test('Should return an empty result when no calendars exist', async () => {
+      const contentfulGraphQLResponse = getContentfulCalendarsGraphqlResponse();
+      contentfulGraphQLResponse.calendarsCollection!.total = 0;
+      contentfulGraphQLResponse.calendarsCollection!.items = [];
+
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+        contentfulGraphQLResponse,
+      );
+
+      const result = await calendarDataProvider.fetch({});
+
+      expect(result).toEqual({
+        items: [],
+        total: 0,
+      });
+    });
+
+    test('Should throw an error with a specific error message when the graphql client throws one', async () => {
+      contentfulGraphqlClientMock.request.mockRejectedValueOnce(
+        new GraphQLError('some error message'),
+      );
+
+      await expect(calendarDataProvider.fetch({})).rejects.toThrow(
+        'some error message',
+      );
+    });
+
+    test('Should return an empty result when the query is returned as null', async () => {
+      const contentfulGraphQLResponse = getContentfulCalendarsGraphqlResponse();
+      contentfulGraphQLResponse.calendarsCollection = null;
+
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+        contentfulGraphQLResponse,
+      );
+
+      const result = await calendarDataProvider.fetch({});
+
+      expect(result).toEqual({
+        items: [],
+        total: 0,
+      });
+    });
+
+    test('Should use default query params when request does not have any', async () => {
+      const contentfulGraphQLResponse = getContentfulCalendarsGraphqlResponse();
+
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+        contentfulGraphQLResponse,
+      );
+
+      const result = await calendarDataProvider.fetch({});
+
+      expect(result).toEqual({
+        total: 1,
+        items: [getContentfulCalendarDataObject()],
+      });
+
+      expect(contentfulGraphqlClientMock.request).toHaveBeenCalledWith(
+        expect.anything(),
```

Comment:
```suggestion
        gp2.FETCH_CALENDARS,
```

I think you can do this when gp2 is exported from '@asap-hub/contentful'